docs/CONTRIBUTING.md

```diff
@@ -1,46 +1,118 @@
+The examples test a sample application called [the internet express](https://github.com/getgauge-contrib/the-internet-express). You can write test scripts for this application or create a self contained working example. 
+
+Please note, examples you submit should not be testing an external website as they are tested on 
+every pull request and this may result in unwanted traffic to the website.
+
+## Improving documentation
+
+Source for Taiko's [documentation](https://docs.taiko.dev) is available on [Github](https://github.com/getgauge/taiko/tree/master/docs). The [API](https://github.com/getgauge/taiko/blob/f98acd6f70ae5c3ad76097455e3f78a9ec461d55/lib/taiko.js#L255) documentation uses [JSDoc](https://github.com/documentationjs/documentation/blob/master/docs/GETTING_STARTED.md#the-essentials). 
+
+Please raise a [pull request](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests) if you want to improve any part of the documentation. A link to a preview site with the changes 
+you made on the pull request will be available under the "All checks have passed" section of the pull request.
+
+## Answering queries
+
+You can help by answering queries by users who are getting started on Taiko or looking for ways to solve specific 
+test automation tasks on [Spectrum](https://spectrum.chat/gauge?tab=posts).
+
+You can also comment on [issues](https://github.com/getgauge/taiko/issues) raised by users. 
+To get notified of new issues make sure are watching the repository.
+
+## Talking/Writing about Taiko
+
+If you feel Taiko can help others in their test automation consider talking about Taiko to your testing community or at other meetups. 
+
+Taiko has great videos on [YouTube](https://www.youtube.com/results?search_query=taiko+testing) created by all kinds of users and there is always scope for more. 
+
+You can also write about Taiko on Gauge's [blog page](https://github.com/getgauge/gauge.org/tree/master/source/posts). If you have an idea please create [Github issue ](https://github.com/getgauge/gauge.org/issues) detailing your idea to discuss the post.
+
+To share your talk or post about Taiko please tweet [@taikodev](https://twitter.com/taikodev). For logos and images please refer Taiko's [brand page](https://brand.taiko.dev)
+
+## Writing plugins
+
+Taiko's [plugin](https://github.com/getgauge/taiko/wiki/Taiko-Plugin) architecture, allows you to 
+extend Taiko's functionality. It is also a way to experiment with ideas which could be a part of the
+core. 
+
+You can refer the current list of [plugins](https://docs.taiko.dev/plugins/) to build your own. If you
+want your plugin to be listed on the [plugins](https://docs.taiko.dev/plugins/) page please modify the
+[list](https://github.com/getgauge/taiko/blob/master/docs/plugins.md) and send a pull request. 
+
+## Contribute code and pull requests to Taiko! 
```

Comment:
Worth referring to some tech docs? We should add those if they don't exist

Reply:
It's there in the para before this one.
```
Taiko's [plugin](https://github.com/getgauge/taiko/wiki/Taiko-Plugin) architecture, allows you to 
 extend Taiko's functionality. It is also a way to experiment with ideas which could be a part of the
 core. Or did you mean something else?
```

Reply:
no that's good enough i think.